ogc/src/main/scala/geotrellis/server/ogc/wms/CapabilitiesView.scala

```diff
@@ -104,6 +115,17 @@ object CapabilitiesView {
         "@resy" -> cellSize.height
       ))
 
+  def timeDimension(times: Seq[Option[ZonedDateTime]]): Seq[Dimension] = {
+    val validTimes = times.flatten
+    if (validTimes.isEmpty) {
+      Nil
+    } else if (validTimes.length == 1) {
+      OgcTimeInterval(validTimes.head)
+    } else {
+      OgcTimeInterval(validTimes.min, Some(validTimes.max), None)
+    }
+  }
```

Comment:
I'm fine with that piece of code

But just to get rid of unsafe `.head` and a length check how about the following:

```scala
validTimes match {
  case Nil         => Nil
  case head :: Nil => OgcTimeInterval(head)
  case list        => OgcTimeInterval(list.min, list.max.some, None) 
}
```

Reply:
TIL that Nil matches an empty Seq:
```scala
scala> Seq() match {
     | case Nil => print("empty")
     | case _ => print("other")
     | }
empty
```

Yeah, this is a much cleaner check. Updated.